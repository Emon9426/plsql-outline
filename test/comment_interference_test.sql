CREATE OR REPLACE procedure comment_test
IS
   /* 多行注释
      可能影响解析 */
   -- 单行注释
   FUNCTION commented_func return varchar2 is
   begin
      return 'test';
   end;
   
   /* 另一个多行注释 */
   FUNCTION func_after_comment(
      /* 参数注释 */
      p1 in varchar2, -- 行末注释
      p2 in number
   )
   return varchar2 is
   begin
      /* 函数体注释 */
      return p1 || to_char(p2);
   end;
BEGIN
   null;
END;
