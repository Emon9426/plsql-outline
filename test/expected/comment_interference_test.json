{
  "nodes": [
    {
      "type": "PROCEDURE",
      "name": "comment_test",
      "declarationLine": 1,
      "beginLine": 22,
      "exceptionLine": null,
      "endLine": 24,
      "level": 1,
      "children": [
        {
          "type": "FUNCTION",
          "name": "commented_func",
          "declarationLine": 6,
          "beginLine": 7,
          "exceptionLine": null,
          "endLine": 9,
          "level": 2,
          "children": []
        },
        {
          "type": "FUNCTION",
          "name": "func_after_comment",
          "declarationLine": 12,
          "beginLine": 18,
          "exceptionLine": null,
          "endLine": 21,
          "level": 2,
          "children": []
        }
      ]
    }
  ],
  "metadata": {
    "sourceFile": "comment_interference_test.sql",
    "totalLines": 17,
    "maxNestingDepth": 2
  }
}